### DISCUSSION TITLE
Using __get_pydantic_core_schema__ in place of _get_validators_ ?

### DISCUSSION THREAD

**Discussion by ed-p-may** [pydantic/pydantic#9562] [Pydantic V2]
Hi, 

I am wondering about how to migrate some old code that uses `_get_validators_` to instead use the `__get_pydantic_core_schema__`, as per the deprecation warning:

> PydanticDeprecatedSince20: `__get_validators__` is deprecated and will be removed, use `__get_pydantic_core_schema__` instead. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.7/migration/

I tried to use the bump-pydantic tool, and it only said that I needed to make the change manually. I have reviewed the [code snippet in the migration guide](https://docs.pydantic.dev/latest/concepts/types/#customizing-validation-with-__get_pydantic_core_schema__), but I admit I am at a loss as to how this compares to my original code?

in my original version, I have a pretty simple setup like this:

```python
class Child(BaseModel):
    . . . 

    @classmethod
    def __get_validators__(cls, *args, **kwargs):
        yield cls.validate

    @classmethod
    def validate(cls, _input, *args, **kwargs):
        # Some work goes on in here...
        . . . 
```

Are there any good resouces I could review that show how to migrate something like this and what an equivalent v2 version would be?

I have also tried using both ChatGPT and GitHub Co-Pilot, but they seem hopelessy confused and mix v1 and v2 syntax and get it all full of errors. Not sure where to turn to on this?

thanks for any suggestions!
@ed-p-may

**Comment by izimbra** [comment]
I'm new to Pydantic but I believe that since `Child` inherits from `BaseModel` you didn't need to implement `__get_validator__` in the first place. This method and its v2 counterpart `__get_pydantic_core_schema__` are meant to be used with custom types that are not Pydantic models.

To validate `Child` instances you can annotate `validate` as `@model_validator` instead, see https://docs.pydantic.dev/latest/concepts/validators/#model-validators
